src/vs/workbench/api/common/extHostDiagnostics.ts

```diff
@@ -117,6 +118,9 @@ export class DiagnosticCollection implements vscode.DiagnosticCollection {
 			}
 		}
 
+		// Don't sync code blocks diagnostics for now but still keep them in the collection
+		toSync = toSync.filter(x => x.scheme !== Schemas.vscodeChatCodeBlock);
```

Comment:
A better place might be this one: https://github.com/microsoft/vscode/blob/a914c155d5ce95793eb70abe5d479665f15aa4c1/src/vs/platform/markers/common/markerService.ts#L15